src/WeightVectors.jl

```diff
@@ -635,7 +635,8 @@ function _set_to_zero!(m::Memory, i::Int)
         else
             m2 = m[2]
             if weight_index == m2 # We zeroed out the first group
-                while chunk == 0 # Find the new m[2]
+                checkbounds(m, level_weights_nonzero_index-10495+5)
+                @inbounds while chunk == 0 # Find the new m[2]
                     level_weights_nonzero_index -= 1
                     m2 -= 64
                     chunk = m[level_weights_nonzero_index]
```

Comment:
This could be out of bounds on the first iteration for corrupt `m`.

Reply:
`level_weights_nonzero_index` is given by `get_level_weights_nonzero_indices(exponent::UInt64)` which is `_convert(Int, 10496 + exponent >> 6)` so it is at least `10496` which means that we could just check `checkbounds(m, level_weights_nonzero_index)`, if this is true memory is not corrupt (at least the loop would work do the same even without inbounds) because we also checked `m[5]` is not zero right before this code block

Reply:
do you agree?